tests/unit/amo/api/test_collections.js

```diff
@@ -298,29 +299,71 @@ describe(__filename, () => {
       };
     };
 
-    it('requires an api parameter', () => {
-      const params = defaultParams();
-      delete params.api;
+    it('validates description value', async () => {
```

Comment:
This test is a duplication of [this one](https://github.com/bobsilverberg/addons-frontend/blob/3668113b1a6dcd16cf3c07b5f75ca9c58cdc3bbf/tests/unit/amo/api/test_collections.js#L368). I see that you're treating `modifyCollection()` as an implementation detail but I'd rather see it treated as a separate, trusted unit of code with its own distinct responsibilities. This will allow us to remove duplicate tests which makes the test suite faster. It also makes the app more maintainable because you'd only have to change one test instead of two if the logic evolves.

To achieve this pattern we need 1) strong types to enforce contracts and 2) a dependency injection mechanism to isolate the unit we're depending on. We already have number 1. Here's a change you can make for the dependency injection:

```js
export const updateCollection = ({
  api,
  collectionSlug,
  defaultLocale,
  description,
  name,
  slug,
  user,
  _modifyCollection = modifyCollection,
  _validateLocalizedString = validateLocalizedString,
}: UpdateCollectionParams): Promise<void> => {
  return _modifyCollection('update', {
    api,
    collectionSlug,
    defaultLocale,
    description,
    name,
    slug,
    user,
    _validateLocalizedString,
  });
};
```

After that, you only need *one* simple test that makes sure `updateCollection` calls a stub of `modifyCollection` with the correct parameters. Flow will enforce the interface contract in a deeper way.

It would then be safe to unit test `modifyCollection` for all of its features because we can trust that `updateCollection` is depending on its interface correctly.